### PR TITLE
Feature 12 support for devices

### LINKED ISSUE #12: Support for device requests

### PR DESCRIPTION
**Pull request by villejuutila**
Resolves #12

### CODE REVIEW

File: src/test/kotlin/fi/metatavu/noheva/api/test/functional/builder/impl/ExhibitionDeviceTestBuilderResource.kt
Comment (anttileppa) [inline src/test/kotlin/fi/metatavu/noheva/api/test/functional/builder/impl/ExhibitionDeviceTestBuilderResource.kt:30]: Shouldn't we always include deviceId when creating exhibition devices?

File: src/main/kotlin/fi/metatavu/noheva/api/DevicesApiImpl.kt
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/noheva/api/DevicesApiImpl.kt:133]: Why do we allow changing approval status from approved to pending?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt]: Resolved.

File: src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt]: Resolved.

File: src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/noheva/devices/DeviceController.kt]: Resolved.

File: src/main/kotlin/fi/metatavu/noheva/persistence/dao/DeviceDAO.kt
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/noheva/persistence/dao/DeviceDAO.kt]: Resolved.

File: src/main/kotlin/fi/metatavu/noheva/api/DevicesApiImpl.kt
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/noheva/api/DevicesApiImpl.kt:133]: I thought that just in case someone would want to revert the approval

**Review by anttileppa — APPROVED** [review]

**Pull request merged**
